core/field_colour.js

```diff
@@ -185,6 +185,26 @@ Blockly.FieldColour.prototype.initView = function() {
   }
 };
 
+/**
+ * Saves this field's value.
+ * @return {string} The colour value held by this field.
+ * @override
+ * @package
+ */
+Blockly.FieldColour.prototype.saveState = function() {
+  return /** @type{string} */ (this.value_);
+};
+
+/**
+ * Sets the field's value based on the given state.
+ * @param {*} state The state to apply to the colour field.
```

Comment:
Could this type be `string`? If we know that we save a string and anything other than a string is invalid for this field?

Reply:
Sadly this cannot be a string :/ The super class accepts `*` so all subclasses must also accept `*`. Here's an example of how this works type-system wise:
```typescript
class AnimalPen {
  constructor() { }

  feed (animal: Animal) {

  }
}

class CowPen extends AnimalPen {
  constructor() { }

  // This is not allowed b/c it is more restrictive.
  feed (cow: Cow) {

  }
}

// CowPen is an AnimalPen so this is legal (ignoring the bad type of feed).
const animalPen: AnimalPen = new CowPen();

const turkey: Turkey = new Turkey();

// Animal pen accepts any animal, so this is legal.
// However, animalPen is actually a CowPen and as such, would blow up.
// Which is why you are not allowed to restrict parameter types in subtypes.
animalPen.feed(turkey);
```

If `Field` was a parameterized type, as outlined in [this issue](https://github.com/google/blockly/issues/5258), we would be able to fix this. But it's not so we can't :/

Reply:
Ah, yup. Sorry wishful thinking : )